core/src/main/java/org/apache/iceberg/avro/PruneColumns.java

```diff
@@ -100,36 +106,46 @@ public Schema union(Schema union, List<Schema> options) {
   }
 
   @Override
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
   public Schema array(Schema array, Schema element) {
-    if (array.getLogicalType() instanceof LogicalMap) {
+    if (array.getLogicalType() instanceof LogicalMap || AvroSchemaUtil.isKeyValueSchema(array.getElementType())) {
       Schema keyValue = array.getElementType();
-      int keyId = AvroSchemaUtil.getFieldId(keyValue.getField("key"));
-      int valueId = AvroSchemaUtil.getFieldId(keyValue.getField("value"));
+      Integer keyId = fieldId(keyValue.getField("key"));
+      Integer valueId = fieldId(keyValue.getField("value"));
+      if (keyId == null) {
+        Preconditions.checkState(valueId == null, "Map schema %s has value id but not key id", array);
```

Comment:
This could happen if the mapping has key, but not value. I'm not sure that it's a good idea to throw an exception in this case. Maybe just warn and not project?

Reply:
+1

Reply:
For now I've added a log warning. Is adding logging for both [key missing and value present and value missing key present] a good idea, or do u see this to be a expected usage in which case I can remove even remove the existing logging